students/shirin_ak/Session05/mailroom.py

```diff
@@ -0,0 +1,165 @@
+               'David':[3, 2500.17],
+               'Michael':[4, 1500.00],
+               'Allen':[1, 2400.35]}
+
+
+
+PROMPT = "\n".join(("Please select a menu option below:",
+                    "1 - Send a Thank You to an individual",
+                    "2 - Create a Report",
+                    "3 - Send letters to all donors",
+                    "4 - Quit",
+                    ">>> "))
+
+
+def create_report():
+    """Create a detailed report of donors and their history of the amounts they have donated"""
+    donor_report = []
+    for name in DONORS_NAME:
+        total_donation = (DONORS_NAME[name][1])
+        donation_times = (DONORS_NAME[name][0])
+        avg_donate = round((total_donation / donation_times), 2) 
+        donor_report.append((name, total_donation, donation_times, avg_donate))
+        donor_report.sort(key=lambda x: x[1], reverse=True)
+    print("\n\n") 
+    print("{:16} {:20} {:25} {:25}\n".format("Name", "Total Donation",
+                                             " Number of Gifts", "Average Gifts"))
+    for i in donor_report: 
+        print("{:16} {:<20} {:<25} {:<25}".format(i[0], i[1], i[2], i[3]))
+    print("\n\n\n")
+
+    
+def add_donor():
+    """ prompt for a Full Name.if user types 'list' show donor names,if types 'quit'
+        return to the main menu ,if types a name not in the list add that to the list,
+        if types a name in the list use it and prompt for donation amount,Add that amount
+        to the donation history """
+
+    while True:
+        print("\nEnter 'list' to see list of names.")
+        print("Enter 'quit' to go back.")
+        print("Enter 'name' to add new donor.")
+        name = input("Please enter a name: ")
+        if name == 'list':
+            show_list()
+        elif name == 'quit':
+            break
+        else:
+            donation_amount = get_donation()
+            if donation_amount == 'quit':
+                break
+            if name not in DONORS_NAME:
+                DONORS_NAME[name] = [1, donation_amount]
+            else:
+                DONORS_NAME[name][0] = DONORS_NAME[name][0] + 1
+                DONORS_NAME[name][1] = DONORS_NAME[name][1] + donation_amount
+            send_thank_you_letter(name, donation_amount)
+            break
+
+        
+def show_list():
+    """ Updated by Comprehensions"""
+    print("\nDonor Names:") 
+    print("-" *15)
+    [print(name) for name in DONORS_NAME]
+
+    
+def get_donation():
+    """ updated this function using Error handling.
+        It gets donation from the donor"""
+    while True:       
+        donation = input("\nPlease Enter Valid Amount."
+                         "\n or Enter 'quit' to return to main menu." "\n$")        
+        if donation == 'quit':
+            return donation        
+        try:
+            donation = int(donation)
+            return float(donation)
+        except ValueError:
+             print("\nError.It is not valid amount.")
```

Comment:
This should be reorganized a bit.  Also testing if a thing can be cast to int, but then casting as float doesn't quite make sense
```
        try:
            donation = input("\nPlease Enter Valid Amount."
                         "\n or Enter 'quit' to return to main menu." "\n$")        
            if donation.lower() == 'quit':
                return donation        
            return float(donation)
        except ValueError:
             print("\nError.It is not valid amount.")
```